plugins/by-name/alpha/default.nix

```diff
@@ -156,20 +165,31 @@ in
       opt = options.plugins.alpha;
     in
     lib.mkIf cfg.enable {
-      # TODO: deprecated 2024-08-29 remove after 24.11
-      warnings = lib.mkIf opt.iconsEnabled.isDefined [
-        ''
-          nixvim (plugins.alpha):
-          The option definition `plugins.alpha.iconsEnabled' in ${lib.showFiles opt.iconsEnabled.files} has been deprecated; please remove it.
-          You should use `plugins.alpha.iconsPackage' instead.
-        ''
-      ];
 
-      extraPlugins =
-        [ cfg.package ]
-        ++ lib.optional (
-          cfg.iconsPackage != null && (opt.iconsEnabled.isDefined -> cfg.iconsEnabled)
-        ) cfg.iconsPackage;
+      # TODO: deprecated 2024-08-29 remove after 24.11
+      warnings =
+        lib.optional opt.iconsEnabled.isDefined [
+          ''
+            nixvim (plugins.alpha):
+            The option definition `plugins.alpha.iconsEnabled' in ${lib.showFiles opt.iconsEnabled.files} has been deprecated; please remove it.
+            Please use `plugins.web-devicons` or `plugins.mini.modules.icons` with `plugins.mini.mockDevIcons` instead.
+          ''
+        ]
+        # TODO: added 2024-09-20 remove after 24.11
+        ++
+          lib.optional
+            (
+              (opt.iconsEnabled.isDefined -> cfg.consEnabled)
+              && options.plugins.web-devicons.enable.highestPrio == 1490
```

Comment:
This doesn't look right.

1. `cfg.consEnabled` looks like a typo
2. The `&&` would mean `plugins.web-devicons.enable`'s priority is only checked when the lhs is true, but we are enabling it below unconditionally

Perhaps this should be two separate warnings? Or an `||` condition?

Reply:
Nah, it should be this condition, but I need to fix the enable below.

Reply:
Perhaps we can nest both the warning and the enable under a single `mkIf` then?

We could always use `mkMerge` if there's other `warnings` defined separately?

Reply:
I didn't realize I had basically duplicate warnings... I was fixing the triggering of the warnings after your comment about the unconditional enabling.

Reply:
I pushed an update that is fixing when it enables the plugin and consolidates the warning.

Reply:
I almost pointed out the other warning checking `isDefined` but figured we'd focus on one thing at a time 😁